apps/tools/src/deployer.command.ts

```diff
@@ -0,0 +1,119 @@
+    const spin = createSpinner()
+    spin.start("Ensuring all relayers can deploy")
+    await Promise.all(
+      this.networkCfg.relayers.map(async (relayer) => {
+        const canDeploy = await this.deployer.canDeploy(relayer)
+        if (canDeploy) {
+          spin.info(`Relayer:${relayer} can deploy ✅`)
+        } else {
+          spin.info(`Relayer:${relayer} can not deploy, updating permissions`)
+          const receipt = await this.deployer.changeDeployerPermission(
+            relayer,
+            true
+          // @ts-ignore
+          ).sendAndWaitForReceipt({
+            from: this.networkCfg.fund.address
+          })
+
+          if (receipt.status === true) {
+            spin.info(`Relayer:${relayer} permissions updated `)
+          } else {
+            spin.fail(`Relayer:${relayer} could not update permissions (tx:${receipt.transactionHash}`)
+          }
+        }
+      })
+    )
+    spin.succeed()
+  }
+
+  @Command({
+    command: 'ensureHasMetaTxWallet',
+    description: 'Ensure that all relayers have a meta-tx wallet deployed',
+    options: [
+      {
+        flags: '-i, --implementation <address>',
+        required: false,
+        defaultValue: null,
+        description: 'The implementation address to use'
+      },
+    ]
+  })
+  async ensureHasMetaTxWallet(cmd: commander.Command): Promise<void> {
+    const opts = cmd.opts()
+    const availableImplementations = Object.keys(
+      this.networkCfg.contracts.MetaTransactionWalletVersions
+    )
+    let implementationAddress: string
+    if (opts.implementation == null) {
+      implementationAddress = availableImplementations[availableImplementations.length - 1]
+    } else {
+      implementationAddress = availableImplementations.find(
+        (impl) => {
+          return normalizeAddress(impl) === normalizeAddress(opts.implementation)
+        }
+      )
+      if (implementationAddress === undefined) {
+        throw(new Error("Invalid implementation address"))
+      }
+    }
+
+
+    const metaTxWallet = newMetaTransactionWallet(this.contractKit.web3, implementationAddress)
+    const spin = createSpinner()
+    spin.start("Ensuring all relayers have wallets")
+
+    await Promise.all(
+      this.networkCfg.relayers.map(async (relayer) => {
+        const wallet = await this.deployer.getWallet(relayer)
+        if (wallet !== NULL_ADDRESS) {
+          spin.info(`Relayer:${relayer} has wallet: ${wallet} ✅`)
+        } else {
+          spin.info(`Relayer:${relayer} needs wallet. Deploying.`)
+          const receipt = await this.deployer.deploy(
+            relayer,
+            implementationAddress,
+            metaTxWallet.methods.initialize(relayer).encodeABI()
+          // @ts-ignore
+          ).sendAndWaitForReceipt({
+            from: this.networkCfg.fund.address
+          })
+
+          if (receipt.status === true) {
+            spin.info(`Relayer:${relayer} wallet deployed!`)
```

Comment:
Perhaps we can also log the tx hash in success case just in case we need to debug